exporters/autoexport/metrics.go

```diff
@@ -28,10 +28,17 @@ type MetricOption = option[metric.Reader]
 
 // WithFallbackMetricReader sets the fallback exporter to use when no exporter
 // is configured through the OTEL_METRICS_EXPORTER environment variable.
-func WithFallbackMetricReader(metricReaderFactory func(ctx context.Context) (metric.Reader, error)) MetricOption {
+func WithFallbackMetricReader(metricReaderFactory func(ctx context.Context, cfg config[metric.Reader]) (metric.Reader, error)) MetricOption {
```

Comment:
It will not be possible to use `WithFallbackMetricReader` as `config[metric.Reader]` is unexported.

Side note: We are missing unit tests that use `WithFallbackMetricReader` and `WithFallbackSpanExporter`